File: CoordinatorLayout/app/src/main/java/com/puresoftware/coordinatorlayout/MainActivity.java
package com.puresoftware.coordinatorlayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LinearLayout lin1 = findViewById(R.id.line1);
        LinearLayout lin2 = findViewById(R.id.line2);
        LinearLayout lin3 = findViewById(R.id.line3);

        lin1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "1번입니다.", Toast.LENGTH_SHORT).show();
            }
        });

        lin2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "2번입니다.", Toast.LENGTH_SHORT).show();
            }
        });

        lin3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "3번입니다.", Toast.LENGTH_SHORT).show();
            }
        });





        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));

        Adapter adapter = new Adapter();
        for (int i = 0; i < 100; i++) {
            String str = i + "번째 아이템";
            adapter.setArrayData(str);
        }
        recyclerView.setAdapter(adapter);
    }
}

File: OverlapRecyclerView/app/src/main/java/com/puresoftware/overlaprecyclerview/MainActivity.kt
package com.puresoftware.overlaprecyclerview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListAdapter
import androidx.recyclerview.widget.LinearLayoutManager
import com.puresoftware.overlaprecyclerview.databinding.ActivityMainBinding
import com.puresoftware.overlaprecyclerview.fragment.MainFragment

// https://kimyunseok.tistory.com/130
class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        supportFragmentManager.beginTransaction().replace(binding.container.id, MainFragment())
            .commit()
    }
}